src/Illuminate/Foundation/Console/TinkerCommand.php

```diff
@@ -26,26 +26,11 @@ class TinkerCommand extends Command {
 	 */
 	public function fire()
 	{
-		if ($this->supportsBoris())
-		{
-			$this->getApplication()->setCatchExceptions(false);
-
-			(new Boris('> '))->start();
-		}
-		else
-		{
-			$this->comment('Boris REPL not supported. Needs readline, posix, and pcntl extensions.');
-		}
-	}
+		$this->getApplication()->setCatchExceptions(false);
 
-	/**
-	 * Determine if the current environment supports Boris.
-	 *
-	 * @return bool
-	 */
-	protected function supportsBoris()
-	{
-		return extension_loaded('readline') && extension_loaded('posix') && extension_loaded('pcntl');
+		$shell = new Shell();
+
+		$shell->run();
```

Comment:
I prefer defining the variable first, but I can change it if you really want @taylorotwell.